gcc/rust/rust-session-manager.cc

```diff
@@ -113,24 +114,32 @@ infer_crate_name (const std::string &filename)
 static bool
 validate_crate_name (const std::string &crate_name, Error &error)
 {
-  if (crate_name.empty ())
+  Utf8String utf8_name = {crate_name};
+  auto uchars = utf8_name.get_chars ();
+
+  if (!uchars.has_value ())
+    {
+      error = Error (UNDEF_LOCATION, "crate name is not a valid UTF-8 string");
+      return false;
+    }
+  if (uchars.value ().empty ())
```

Comment:
maybe it would be clearer to add a temporary variable after the first `if (!uchars.has_value())`? You keep calling `uchars.value()` which maybe hurts readability just a little bit?

Reply:
Sounds good. fixed